File: faq-plus/bot/models/changeTicketStatusPayload.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

/**
 * Represents the data payload of Action.Submit to change the status of a ticket.
 */
export class ChangeTicketStatusPayload {
  /**
   * Action that reopens a closed ticket.
   */
  public static readonly reopenAction: string = "Reopen";

  /**
   * Action that closes a ticket.
   */
  public static readonly closeAction: string = "Close";

  /**
   * Action that assigns a ticket to the person that performed the action.
   */
  public static readonly assignToSelfAction: string = "AssignToSelf";

  /**
   * The ticket id.
   */
  public ticketId: string;

  /**
   * The action to perform on the ticket.
   */
  public action?: string;
}

File: faq-plus/cofigapp/Microsoft.Teams.Apps.FAQPlusPlus.Common/Models/ChangeTicketStatusPayload.cs
// <copyright file="ChangeTicketStatusPayload.cs" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

namespace Microsoft.Teams.Apps.FAQPlusPlus.Common.Models
{
    using Newtonsoft.Json;

    /// <summary>
    /// Represents the data payload of Action.Submit to change the status of a ticket.
    /// </summary>
    public class ChangeTicketStatusPayload
    {
        /// <summary>
        /// Action that reopens a closed ticket.
        /// </summary>
        public const string ReopenAction = "Reopen";

        /// <summary>
        /// Action that closes a ticket.
        /// </summary>
        public const string CloseAction = "Close";

        /// <summary>
        /// Action that assigns a ticket to the person that performed the action.
        /// </summary>
        public const string AssignToSelfAction = "AssignToSelf";

        /// <summary>
        /// Gets or sets the ticket id.
        /// </summary>
        [JsonProperty("ticketId")]
        public string TicketId { get; set; }

        /// <summary>
        /// Gets or sets the action to perform on the ticket.
        /// </summary>
        [JsonProperty("action")]
        public string Action { get; set; }
    }
}